File: tests/python/cases/case_112_deferred_sink_gate_02.py
"""
Task case 112: deferred_sink_gate_02 false-positive/false-negative stress pattern.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_112_deferred_sink_gate_02(input_data):
    task_name = 'deferred_sink_gate_02'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'stress',
        'stage': 'evaluation',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    if raw.startswith('x'):
        # ESCAPE: conditional sink persists payload only on specific branch.
        RETAINED_AUDIT.append(payload)
    return 'ok'

File: tests/go/cases/case_112_deferred_sink_gate_02.go
package escape_tests

var retainedCase112 = []map[string]string{}

func Case112DeferredSinkGate02(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "deferred_sink_gate_02",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "deferred_sink_gate_02:" + raw,
	}
	if len(raw) > 0 && raw[0] == 'x' {
		// ESCAPE: conditional path retention.
		retainedCase112 = append(retainedCase112, payload)
	}
	return "ok"
}
